src/device_builder.rs

```diff
@@ -106,27 +108,132 @@ impl<'a, B: UsbBus> UsbDeviceBuilder<'a, B> {
         self
     }
 
+    /// Sets **extra** Language ID for device.
+    ///
+    /// Since "en_US"(0x0409) is implicitly embedded, you just need to fill other LangIDs
+    ///
+    /// Default: (none)
+    pub fn set_extra_lang_ids(mut self, extra_lang_ids: &'a [LangID]) -> Self {
+        if extra_lang_ids.len() == 0 {
+            self.config.extra_lang_ids = None;
+            return self;
+        }
+
+        assert!(
+            extra_lang_ids.len() < 16,
+            "Not support more than 15 extra LangIDs"
+        );
+
+        [
+            self.config.manufacturer,
+            self.config.product,
+            self.config.serial_number,
+        ]
+        .iter()
+        .zip(["manufacturer", "product", "serial_number"].iter())
+        .for_each(|(list, field_name)| {
+            // do list length check only if user already specify "manufacturer", "product" or "serial_number"
+            if let Some(list) = list {
+                assert!(
+                    extra_lang_ids.len() == list.len() - 1,
+                    "The length of \"extra_lang_id\" list should be one less than \"{}\" list",
+                    field_name
+                )
+            }
+        });
+
+        self.config.extra_lang_ids = Some(extra_lang_ids);
+
+        self
+    }
+
     /// Sets the manufacturer name string descriptor.
     ///
+    /// the first string should always be in English, the language of rest strings
+    /// should be pair with what inside [.extra_lang_ids()](Self::extra_lang_ids)
+    ///
     /// Default: (none)
-    pub fn manufacturer(mut self, manufacturer: &'a str) -> Self {
-        self.config.manufacturer = Some(manufacturer);
+    pub fn manufacturer(mut self, manufacturer_ls: &'a [&'a str]) -> Self {
+        if manufacturer_ls.len() == 0 {
+            self.config.manufacturer = None;
+            return self;
+        }
+
+        assert!(
+            manufacturer_ls.len() <= 16,
+            "Not support more than 16 \"manufacturer\"s"
+        );
+
+        // do list length check only if user already specify "extra_lang_ids"
+        if let Some(extra_lang_ids) = self.config.extra_lang_ids {
+            assert!(
+                manufacturer_ls.len() == extra_lang_ids.len() + 1,
+                "The length of \"manufacturer\" list should be one more than \"extra_lang_ids\" list",
+            )
+        }
```

Comment:
This check needs to be done even if there are no extra lang IDs. I.e.
```suggestion
        let num_extra_langs = self.config.extra_lang_ids.as_ref().map(|langs| langs.len()).unwrap_or(0);
        
        assert!(
            manufacturer_ls.len() == num_extra_langs + 1,
            "The length of \"manufacturer\" list should be one more than \"extra_lang_ids\" list",
        )

```

Reply:
I re-think about this. And I think this code may force user call `.set_extra_lang_ids()` before other methods.

This code will panic

```rust
builder.product(&["name1", "name2"]).set_extra_ids(&[lang_id1])
```

If we need this, we may doc the "calling" order of the methods.

Reply:
Or use a different builder type, e.g. with a generic marker.

Reply:
Ah you're right @eZioPan. In general, I think the builder needs to change like @ithinuel notes, but I'd prefer if we did that in a different PR. Let's at least get _something_ in here :)